resource/plugins/unreal/python/publisher/post_finalizers/unreal_dependencytrack_publisher_post_finalizer.py

```diff
@@ -0,0 +1,137 @@
+        loader defined in *options*.'''
+
+        import json
+
+        # Extract version ID from the run data
+        asset_version_id = component_name = asset_filesystem_path = None
+        for comp in data:
+            if comp['type'] == core_constants.COMPONENT:
+                for result in comp['result']:
+                    if result['name'] == core_constants.EXPORTER:
+                        plugin_result = result['result'][0]
+                        asset_filesystem_path = plugin_result['result'][0]
+                        break
+            elif comp['type'] == core_constants.FINALIZER:
+                for result in comp['result']:
+                    if result['name'] == core_constants.FINALIZER:
+                        plugin_result = result['result'][0]
+                        asset_version_id = plugin_result['result'][
+                            'asset_version_id'
+                        ]
+                        for component_name in plugin_result['result'][
+                            'component_names'
+                        ]:
+                            if not component_name in [
+                                'thumbnail',
+                                'reviewable',
+                            ]:
+                                break
+                        break
+
+        if 'asset_path' in options:
+            # Given pre fetched by batch publisher
+            asset_path = options['asset_path']
+            param_dict = options['param_dict']
+            dcc_object_name = options['dcc_object_name']
+        else:
+            if not asset_filesystem_path:
+                return {'message': 'No asset could be extracted from publish!'}
+
+            # Convert to game path
+            asset_path = unreal_utils.filesystem_asset_path_to_asset_path(
+                asset_filesystem_path
+            )
+
+            # Check if already has snapshot asset info
+            dcc_object_name, param_dict = unreal_utils.get_asset_info(
+                asset_path, snapshot=True
+            )
+
+        if param_dict:
+            # Store the asset info as metadata in ftrack
+            self.logger.warning(
+                'Removing existing snapshot asset info @ "{}"!'.format(
+                    param_dict
+                )
+            )
+            unreal_utils.delete_ftrack_node(dcc_object_name)
+
+        # Create new snapshot asset info
+        component = self.session.query(
+            'Component where version.id is {} and name is "{}"'.format(
+                asset_version_id, component_name
+            )
+        ).one()
+        # Find out were the asset got stored
+        location = self.session.pick_location()
+        component_path = location.get_filesystem_path(component)
+        ftrack_object_manager = self.FtrackObjectManager(self.event_manager)
+        ftrack_object_manager.asset_info = (
```

Comment:
We should directly generate the asset info here using the methods that we already have. Wait for the https://app.clickup.com/t/865bk0ag8 task to be done before.